application/src/test/java/org/thingsboard/server/controller/AbstractWebTest.java

```diff
@@ -330,18 +333,39 @@ public void teardownWebTest() throws Exception {
         log.debug("Executing web test teardown");
 
         loginSysAdmin();
-        doDelete("/api/tenant/" + tenantId.getId().toString())
-                .andExpect(status().isOk());
+        deleteTenant(tenantId);
         deleteDifferentTenant();
-
         verifyNoTenantsLeft();
 
         tenantProfileService.deleteTenantProfiles(TenantId.SYS_TENANT_ID);
 
         log.info("Executed web test teardown");
     }
 
-    void verifyNoTenantsLeft() throws Exception {
+    private void verifyNoTenantsLeft() throws Exception {
+        List<Tenant> loadedTenants = getAllTenants();
+        if (!loadedTenants.isEmpty()) {
+            loadedTenants.forEach(tenant -> deleteTenant(tenant.getId()));
```

Comment:
I suggest you log a warning if loadedTenants is not empty. this may help to find the root cause test (first match).

Reply:
We should add waring every time we can't delete the tenant and also add a sleep for 1 sec between retries